node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/io/FileSystem.java

```diff
@@ -36,26 +37,63 @@ public void createDirectory(Path path, FileAttribute<?>... attributes) {
         uncheck(() -> Files.createDirectory(path, attributes));
     }
 
+    public String readUtf8File(Path path) {
+        byte[] byteContent = uncheck(() -> Files.readAllBytes(path));
+        return new String(byteContent, StandardCharsets.UTF_8);
+    }
+
     public void writeUtf8File(Path path, String content, OpenOption... options) {
         byte[] contentInUtf8 = content.getBytes(StandardCharsets.UTF_8);
         uncheck(() -> Files.write(path, contentInUtf8, options));
     }
 
+    private PosixFileAttributes getAttributes(Path path) {
+        return uncheck(() ->
+                Files.getFileAttributeView(path, PosixFileAttributeView.class).readAttributes());
+    }
+
+    public Set<PosixFilePermission> getPermissions(Path path) {
```

Comment:
Consider returning `String` so it matches the set counterpart and simplifies testing.

Reply:
Done